File: dotnet-nunit-selenium/Pages/FormPage.cs
using Selenium.WebDriver.SelfHealing;
using OpenQA.Selenium;

namespace SelfHealingLocatorsDemo;

class FormPage : BasePage
{
    By _textMessageLocator = By.XPath("//p[1] | //*[@id='displayMessage']");
    By _inputTextLocator;
    By _updateButtonLocator; 
    string _classification;

    public FormPage(string url, By inputTextLocator, By updateButtonLocator, string classification, ISelfHealingWebDriver webDriver) 
    : base(webDriver)
    {
        this.WebDriver.Navigate().GoToUrl(url);
        _inputTextLocator = inputTextLocator;
        _updateButtonLocator = updateButtonLocator;
        _classification = classification;
    }

    internal string GetLabel() => WebDriver.FindElement(_textMessageLocator).Text;

    internal void UpdateLabel(string expectedMessage)
    {
        IWebElement inputElement = WebDriver.FindElement(_inputTextLocator);
        inputElement.SendKeys(expectedMessage);
        IWebElement updateButton = WebDriver.FindElement(_updateButtonLocator);
        updateButton.Click();
    }
}

File: java-testng-selenium/src/test/java/selfhealinglocators/Pages/FormPage.java
package selfhealinglocators.Pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class FormPage extends PageBase {
    
    By textMessageLocator = By.xpath("//p[1] | //*[@id='displayMessage']");
    By inputTextLocator;
    By updateButtonLocator; 

    public FormPage(String url, By inputTextLocator, By updateButtonLocator, WebDriver webDriver){
        super(webDriver);
        webDriver.navigate().to(url);
        this.inputTextLocator = inputTextLocator;
        this.updateButtonLocator = updateButtonLocator;
    }

    public String getLabel()
    {
        return webDriver.findElement(textMessageLocator).getText();
    }

    public void updateLabel(String expectedMessage)
    {
        WebElement inputElement = webDriver.findElement(inputTextLocator);
        inputElement.sendKeys(expectedMessage);
        WebElement updateButton = webDriver.findElement(updateButtonLocator);
        updateButton.click();
    }

}
